tests/unit/plugins/module_utils/test_xml_utils.py

```diff
@@ -317,18 +324,16 @@ def test_etree_to_dict__simple_tree(etree_root: Element) -> None:
     - Expected Output: {"foo": [{ "bar" : 1 }, {"bar": 2 }, {"bar" :3 }]}
     """
     output_dict: dict = xml_utils.etree_to_dict(etree_root)
-
-    assert isinstance(output_dict["foo"], list)
-    assert len(output_dict["foo"]) == len(list(etree_root))
-
-    input_children: List[Element] = list(etree_root)
-    for child in input_children:
-        assert any(list(filter(lambda i: child.tag in list(i.keys()), output_dict["foo"])))
+    assert output_dict == {"foo": [{"bar": 1}, {"bar": 2}, {"bar": 3}]}
```

Comment:
This output_dict matches the the docstring example, however the provided xml data from the `pytest.mark.parametrize` decorator injects two xmlstrings which can be matched with the following corrected assertion:

 ```suggestion
     assert isinstance(output_dict["foo"], list)
        assert len(output_dict["foo"]) == len(list(etree_root))
    
        input_children: List[Element] = list(etree_root)
        for child in input_children:
            assert any(list(filter(lambda i, c=child: c.tag in list(i.keys()), output_dict["foo"])))
```

Reply:
couldn't apply --> outdated. But it's fixed: https://github.com/puzzle/puzzle.opnsense/pull/35/commits/9edfe9939efe70432e46fec5382d5b261b6bbdf1